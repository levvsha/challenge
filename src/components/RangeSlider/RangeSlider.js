import 'react-input-range/lib/css/index.css';
import './RangeSlider.styl';

import React, { Component } from 'react';
import Type from 'prop-types';
import InputRange from 'react-input-range';

export default class RangeSlider extends Component {
  static propTypes = {
    extremums: Type.arrayOf(Type.number).isRequired,
    label: Type.string,
    name: Type.string.isRequired,
    onChange: Type.func.isRequired,
    value: Type.number.isRequired
  };

  handleChange = (value) => {
    this.props.onChange(value, this.props.name);
  }

  render() {
    const {
      extremums,
      label,
      value
    } = this.props;

    return (
      <div className="c-range-slider">
        <div className="label">
          {label}
        </div>
        <div className="range-slider">
          <InputRange
            maxValue={extremums[1]}
            minValue={extremums[0]}
            onChange={this.handleChange}
            value={value}
          />
        </div>
      </div>
    );
  }
}
