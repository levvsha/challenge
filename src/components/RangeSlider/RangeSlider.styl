.c-range-slider {
  user-select: none

  .label {
    font-size: 1.6rem
    font-weight: 700
    margin-bottom: 2rem
  }

  .input-range__slider {
    background-color: $mainColor
    border-color: $mainColor
  }

  .input-range__track--active {
    background-color: $mainColor
  }

  .input-range__label--value {
    font-size: 1.6rem
    font-weight: 700

    .input-range__label-container {
      top: -13px
    }
  }

  .range-slider {
    margin-bottom: 35px
  }
}