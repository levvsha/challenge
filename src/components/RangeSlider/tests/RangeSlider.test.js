import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import RangeSlider from '../';

describe('<RangeSlider />', () => {
  it('RangeSlider renders correctly', () => {
    const tree = renderer.create(
      <RangeSlider
        name="width"
        value={4}
        label="Set maze width:"
        extremums={[1, 10]}
        onChange={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('RangeSlider onChange handler is triggered', () => {
    const updateSettingsSpy = jest.fn();

    const renderedComponent = shallow(
      <RangeSlider
        name="width"
        value={4}
        label="Set maze width:"
        extremums={[1, 10]}
        onChange={updateSettingsSpy}
      />
    );

    renderedComponent.find('InputRange').prop('onChange')();

    expect(updateSettingsSpy).toHaveBeenCalled();
  });
});

