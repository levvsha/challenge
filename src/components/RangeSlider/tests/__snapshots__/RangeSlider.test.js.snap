// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSlider /> RangeSlider renders correctly 1`] = `
<div
  className="c-range-slider"
>
  <div
    className="label"
  >
    Set maze width:
  </div>
  <div
    className="range-slider"
  >
    <div
      aria-disabled={false}
      className="input-range"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <span
        className="input-range__label input-range__label--min"
      >
        <span
          className="input-range__label-container"
        >
          1
        </span>
      </span>
      <div
        className="input-range__track input-range__track--background"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="input-range__track input-range__track--active"
          style={
            Object {
              "left": "0%",
              "width": "33.33333333333333%",
            }
          }
        />
        <span
          className="input-range__slider-container"
          style={
            Object {
              "left": "33.33333333333333%",
              "position": "absolute",
            }
          }
        >
          <span
            className="input-range__label input-range__label--value"
          >
            <span
              className="input-range__label-container"
            >
              4
            </span>
          </span>
          <div
            aria-valuemax={10}
            aria-valuemin={1}
            aria-valuenow={4}
            className="input-range__slider"
            draggable="false"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            role="slider"
            tabIndex="0"
          />
        </span>
      </div>
      <span
        className="input-range__label input-range__label--max"
      >
        <span
          className="input-range__label-container"
        >
          10
        </span>
      </span>
    </div>
  </div>
</div>
`;
